src/lib/containers/QueryWrapper.tsx

```diff
@@ -156,72 +77,13 @@ export function QueryWrapper(props: QueryWrapperProps) {
 
   const { data: entity } = useGetEntity<Table>(entityId, versionNumber)
 
-  const [currentPage, setCurrentPage] = useState<number | 'ALL'>(0)
-
-  async function appendNextPageToResults(): Promise<void> {
-    if (!hasNextPage) {
-      throw new Error(
-        'Called appendNextPageToResults when there is no next page',
-      )
-    }
-    await fetchNextPage()
-    setCurrentPage('ALL')
-  }
-
-  async function goToNextPage(): Promise<void> {
-    if (!hasNextPage) {
-      throw new Error('Called goToNextPage when there is no next page')
-    }
-    if (currentPage === 'ALL') {
-      throw new Error('Cannot go to next page when all pages are displayed')
-    }
-    await fetchNextPage()
-    setCurrentPage(currentPage + 1)
-  }
-
-  const hasPreviousPage = currentPage !== 'ALL' && currentPage > 0
-
-  async function goToPreviousPage(): Promise<void> {
-    if (currentPage === 'ALL') {
-      throw new Error('Cannot go to previous page when all pages are displayed')
-    }
-    if (!hasPreviousPage) {
-      throw new Error('Called goToNextPage when there is no next page')
-    }
-
-    await fetchPreviousPage()
-    setCurrentPage(currentPage - 1)
+  const goToPage = async (pageNum: number) => {
+    const lastQueryRequestDeepClone = getLastQueryRequest()
+    lastQueryRequestDeepClone.query.offset =
+      (pageNum - 1) * (lastQueryRequest.query.limit ?? DEFAULT_PAGE_SIZE)
+    executeQueryRequest(lastQueryRequestDeepClone)
   }
```

Comment:
As I mentioned in QueryContext, let's determine the page number here, so it's tied to query state rather than UI state.

Not sure if this math is correct, hopefully you get the idea.

```suggestion
  const pageSize = lastQueryRequest.query.limit ?? DEFAULT_PAGE_SIZE
  
  const currentPageNumber = Math.ceil((lastQueryRequestDeepClone.query.offset + pageSize) / pageSize)
  
  const goToPage = async (pageNum: number) => {
    const lastQueryRequestDeepClone = getLastQueryRequest()
    lastQueryRequestDeepClone.query.offset =
      (pageNum - 1) * pageSize
    executeQueryRequest(lastQueryRequestDeepClone)
  }
```